lib/PEX.ts

```diff
@@ -394,12 +412,47 @@ export class PEX {
     return result;
   }
 
-  private static allowMultipleVCsPerPresentation(verifiableCredential: Array<OriginalVerifiableCredential>) {
-    return !verifiableCredential.some(
-      (c) => CredentialMapper.isJwtEncoded(c) || CredentialMapper.isSdJwtEncoded(c) || CredentialMapper.isSdJwtDecodedCredential(c),
-    );
-  }
+  private static allowMultipleVCsPerPresentation(verifiableCredentials: Array<OriginalVerifiableCredential>): boolean {
+    const jwtCredentials = verifiableCredentials.filter(
+      (c) => CredentialMapper.isJwtEncoded(c) || CredentialMapper.isJwtDecodedCredential(c)
+    )
+
+    if (jwtCredentials.length > 0) {
+      const subjects = new Set<string>()
+      const verificationMethods = new Set<string>()
+
+      for (const credential of jwtCredentials) {
+        const decodedCredential = CredentialMapper.isJwtEncoded(credential) 
+          ? CredentialMapper.decodeVerifiableCredential(credential) as JwtDecodedVerifiableCredential
+          : credential as JwtDecodedVerifiableCredential
+
+        const subject = decodedCredential.sub || (decodedCredential.vc && 'id' in decodedCredential.vc.credentialSubject && decodedCredential.vc.credentialSubject.id);
+        if (subject) {
+          subjects.add(subject);
+        }
 
+        const vcProof = decodedCredential.proof ?? decodedCredential.vc.proof;
+        const proofs = Array.isArray(vcProof) ? vcProof : [vcProof];
```

Comment:
A JWT VP could also contain json-ld credentials.
JWT decoded credentials do not have a proof/verificationMethod (that is only something internal to our agent/wallet). Once you call the decode function on the mapper it will not have the proof anymore. At the same time as mentioned json-ld VCs could be part of a JWT VP, so the logic can remain, but the above filter for jwt only then should be broadened.

This whole approach need to change anyway. We really should map all the VCs onto prospective presentations first, combining what can be combined. But lets do that in a seperate PR as I am creating Credential Info/constrains objects anyway that would halp here.